docs/how-tos/create-soap-endpoint.rst

```diff
@@ -0,0 +1,193 @@
+       <soapenv:Header/>
+       <soapenv:Body>
+         <mySOAPContent>
+           My SOAP Content
+         </mySOAPContent>
+       </soapenv:Body>
+     </soapenv:Envelope>' \
+     --header 'Content-Type: text/xml' \
+     --header 'Accept: text/xml' \
+     http://localhost:3000/rpc/my_soap_endpoint
+
+The output should contain the original request body within the ``therequestbodywas`` entity,
+and should roughly look like:
+
+.. code-block:: xml
+
+    <soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
+      <soapenv:Header/>
+      <soapenv:Body>
+        <therequestbodywas>
+          <soapenv:Body xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
+            <mySOAPContent>
+              My SOAP Content
+            </mySOAPContent>
+          </soapenv:Body>
+        </therequestbodywas>
+      </soapenv:Body>
+    </soapenv:Envelope>
+
+Unforunately the ``Accept: text/xml`` header is strictly required, otherwise postgrest will respond
+with a ``Content-Type: application/json`` header and enclose the response with quotes.
+(You can check the returned headers by adding ``-v`` to the curl call.)
+
+If your SOAP clients do not send the ``Accept: text/xml`` header, you can fix that in your nginx reverse proxy
+by adding something like ...
+
+.. code-block:: nginx
+
+    set $accept $http_accept;
+    if ($contentType ~ "^text/xml($|;)") {
+        set $accept "text/xml";
+    }
+    proxy_set_header Accept $accept;
+
+to your ``location`` nginx configuration.
+(The given example sets the ``Accept`` header for each request of Content-Type ``text/xml``.)
+
+
+A more elaborate example
+------------------------
+
+Adding error handling:
+
+
+.. code-block:: postgres
+
+   -- helper function
+   CREATE OR REPLACE FUNCTION _soap_envelope(body xml)
+    RETURNS xml
+    LANGUAGE sql
+   AS $function$
+     SELECT xmlelement(
+       NAME "soapenv:Envelope",
+       XMLATTRIBUTES('http://schemas.xmlsoap.org/soap/envelope/' AS "xmlns:soapenv"),
+       xmlelement(NAME "soapenv:Header"),
+       xmlelement(NAME "soapenv:Body", body)
+     );
+   $function$;
+   
+   -- helper function
+   CREATE OR REPLACE FUNCTION _soap_exception(
+     faultcode text,
+     faultstring text
+   )
+    RETURNS xml
+    LANGUAGE sql
+   AS $function$
+     SELECT _soap_envelope(
+       xmlelement(NAME "soapenv:Fault",
+         xmlelement(NAME "faultcode", faultcode),
+         xmlelement(NAME "faultstring", faultstring)
+       )
+     );
+   $function$;
+   
+   CREATE OR REPLACE FUNCTION my_soap_endpoint(xml)
+    RETURNS xml
+    LANGUAGE plpgsql
+   AS $function$
+   DECLARE
+     nsarray CONSTANT text[][] := ARRAY[
+       ARRAY['soapenv', 'http://schemas.xmlsoap.org/soap/envelope/']
+     ];
+     exc_msg text;
+     exc_detail text;
+     exc_hint text;
+     exc_sqlstate text;
+   BEGIN
+     -- simulating a statement that results in an exception:
+     RETURN _soap_endpoint((1 / 0)::xml);
```

Comment:
Hm, this `_soap_endpoint` function doesn't exist. Was this on purpose or perhaps you meant to reference `my_soap_endpoint`?